source/camera/SlopeWalk.cpp

```diff
@@ -0,0 +1,294 @@
+{
+    return mVertices[mTopRightIndex];
+}
+
+VectorInt64& SlopeWalk::getBottomRightVertex()
+{
+    return mVertices[mDownRightIndex];
+}
+
+
+
+void SlopeWalk::printState()
+{
+    std::cerr << "mLeftVertices" << std::endl;
+    for(auto ii = mLeftVertices.begin(); ii != mLeftVertices.end(); ii++)
+        std::cerr << mVertices[*ii] << std::endl;
+    std::cerr << "mRightVertices" << std::endl;
+    for(auto ii = mRightVertices.begin(); ii != mRightVertices.end(); ii++)
+        std::cerr << mVertices[*ii] << std::endl;
+}
+
+void SlopeWalk::findMinMaxLeft(const std::vector<VectorInt64> &aa)
+{
+
+    auto min = aa.begin();
+    auto max = aa.begin();
+    for(auto ii = aa.begin(); ii !=aa.end(); ii++ )
+    {
+        if(ii->y < min->y)
+            min = ii;
+        if(ii->y >= max->y)
+            max = ii;
+
+    }
+     mTopLeftIndex = max - aa.begin();
+     mDownLeftIndex = min - aa.begin();
+}
+
+void SlopeWalk::findMinMaxRight(const std::vector<VectorInt64> &aa)
+{
+    auto min = aa.begin();
+    auto max = aa.begin();
+    for(auto ii = aa.begin(); ii !=aa.end(); ii++ )
+    {
+        if(ii->y <= min->y)
+            min = ii;
+        if(ii->y > max->y)
+            max = ii;
+    }
+    mTopRightIndex = max - aa.begin();
+    mDownRightIndex = min - aa.begin();
+}
+
+
+std::string SlopeWalk::debug()
+{
+
+    std::stringstream ss;
+    ss << " mTopLeftIndex " <<  mTopLeftIndex << " mTopRightIndex " << mTopRightIndex << " mDownLeftIndex " << mDownLeftIndex << " mDownRightIndex" << mDownRightIndex << std::endl;
+    
+    for (unsigned int ii = 0 ; ii < mVertices.mMyArray.size() ; ii++ )
+    {
+        VectorInt64 foobar = mVertices[ii];
+        ss<< ii << " " <<double(foobar.x)/VectorInt64::UNIT << " " << double(foobar.y)/VectorInt64::UNIT <<  std::endl;
+    }
+ 
+    ss << "mLeftVertices" << std::endl;
+
+    for(auto ii : mLeftVertices)
+        ss << ii << " " ;
+    ss << std::endl ;
+    ss << "mRightVertices" << std::endl;
+
+    for(auto ii : mRightVertices)
+        ss << ii << " " ;    
+    ss << std::endl ; 
+
+    ss<< "mRightSlopes " << std::endl;
+    for(auto ii : mRightSlopes)
+        ss << double(ii)/VectorInt64::UNIT << std::endl;
+    ss << std::endl; 
+
+    ss<< "mLeftSlopes " << std::endl;    
+    for(auto ii : mLeftSlopes)
+        ss << double(ii)/VectorInt64::UNIT << std::endl;
+
+    return ss.str();
+}
+
+void SlopeWalk::convex_hull(){
+    const double zoomFactorValue = 1.0565;
+    mVertices.zoom(zoomFactorValue);
+    Polygon polygon;
+    boost::geometry::append(polygon, mVertices.mMyArray);
+
+    assert(boost::geometry::num_points(polygon) == 8);
+
+    boost::geometry::correct(polygon);
+
+    assert(boost::geometry::num_points(polygon) == 9);
+
+    Polygon hull;
+    mVertices.mMyArray.clear();
+
+    boost::geometry::convex_hull(polygon, hull);
+    boost::geometry::for_each_point(boost::geometry::exterior_ring(hull), [this](Point &p){this->mVertices.mMyArray.push_back(p);});
```

Comment:
For the next time, please split this into several lines to make clear there is a function here. Not blocking